### ISSUE TITLE
woocommerce_cart_item_name might break new function for aria-label

### ISSUE THREAD

**Issue by SiKth** [woothemes/woocommerce#38795]
Hi,
This update might have caused and issue with the hook "woocommerce_cart_item_name". A lot of people use this to add information after the product name in the Cart and in the Checkout. Now all that data also gets output in the aria-label field. I have this on my site and when I search for this a lot of people recommend this solution to do just this. Anything we can look into?

Try this code for example, where the data comes from custom fields.

```
`add_filter( 'woocommerce_cart_item_name', 'cfwc_cart_item_name', 10, 3 );
function cfwc_cart_item_name( $name, $cart_item, $cart_item_key ) {
	
	if( isset( $cart_item['size'] ) ) {
		$name .= '<br><b>Size:</b> ' . $cart_item['size'] . '';
	}
	if( isset( $cart_item['color'] ) ) {
		$name .= '<br><b>Color:</b> ' . $cart_item['color'] . '';
	}
	
	return $name;
}`
```

_Originally posted by @SiKth in https://github.com/woocommerce/woocommerce/issues/37830#issuecomment-1596235530_

**Comment by SiKth** [comment]
The woocommerce_after_cart_item_name might be a solution but it only works in the cart and not in the checkout. 

Right now I have solved it by using _woocommerce_after_cart_item_name_ for the cart. 
And _woocommerce_cart_item_name_ with the add of `if ( is_checkout() ) {}` for the checkout.
And _woocommerce_order_item_name_ for the order and mail.

Before the _woocommerce_cart_item_name_  was the only one that was needed for this and my concern is that this is mentioned a lot in different forums to change or edit product names or add custom meta after the product name.

**Comment by ishwar-singh-solanki** [comment]
hi,
i want to point out a second issue related to the same hook,
this filter hook "woocommerce_cart_item_name" is used twice in the cart page.
in template/cart/cart.php line 49 to filter the product name
`$product_name = apply_filters( 'woocommerce_cart_item_name', $_product->get_name(), $cart_item, $cart_item_key );`

this filtered product name is used in remove cart area label which may break the area label if it contain HTML content.

then again on line 96/104 (if statement) with the previously filtered product name as input value
`echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', $product_name, $cart_item, $cart_item_key ) . '&nbsp;' );`
`echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', sprintf( '<a href="%s">%s</a>', esc_url( $product_permalink ), $product_name ), $cart_item, $cart_item_key ) );`

this causes the product name to filter twice and add the same content two times, so if you want to add any content to the product name it will add up twice

**Comment by mpandzo** [comment]
@ishwar-singh-solanki are you using `woocommerce_after_cart_item_name` with WooCommerce blocks? Or with the legacy cart and checkout shortcodes?

**Comment by ishwar-singh-solanki** [comment]
no, I am not using the woocommerce_after_cart_item_name hook, I need to show an additional label with certain products (HTML code) in the cart, checkout, and review order pages, and the woocommerce_cart_item_name hook works fine everywhere just that in the cart it is used twice and causing issue of multiple label.

**Comment by mpandzo** [comment]
@ishwar-singh-solanki I have not been able to use the `woocommerce_cart_item_name` with WooCommerce blocks. With classic `woocommerce_cart` and `woocommerce_checkout` shortcode it's fine.

**Comment by pierorocca** [comment]
Hi @SiKth - thanks for reporting this, and for the detail you gave.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed, or it's no longer happening on trunk.

Here's what I found: Current trunk sanitizes the aria-label source with wp_strip_all_tags and does not pass the already filtered $product_name back through woocommerce_cart_item_name in the linked branch. The two concrete regressions described in the issue thread are addressed on trunk.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.

Fix/changelog reference: Current cart template strips tags for the remove-link aria-label and avoids re-filtering an already filtered product name in the linked-name branch.
